### ISSUE TITLE
[SMARTSCAN] [HOLD for payment 2023-11-02] [$500] LOW: Keyboard shouldn't be active/enabled automatically for request details

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#26702]
We automatically enable the mobile keyboard for requests, which is inconsistent with the behavior for reports. Given that a comment is completely optional, let's bring request details inline with reports and not enable the keyboard until the comment input is tapped. 

Example of the inconsistency in behavior (note: I specifically tapped in the report comment input, but did not do the same for the request details, yet the keyboard was automatically enabled. 

https://github.com/Expensify/App/assets/1127863/22fea4ba-c391-4ad6-a972-15ba3e073d98

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011deba88779e69bde</li>
        <li>Upwork Job ID: 1698762396132085760</li>
        <li>Last Price Increase: 2023-09-04</li>
<li>Automatic offers: </li>
<ul>
<li>Pujan92 | Contributor | 26521898</li>
</ul></li>
    </ul>
</details>

**Comment by Pujan92** [comment]
I think it is intentional to focus the input when the report chat is empty. 

https://github.com/Expensify/App/blob/d2bdaece9151c8c65d3b4bc4c9578be663f1950d/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L113-L114

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Distance Request Details Page opening keyborad automatically which is not required.

### What is the root cause of that problem?
- `isEmptyChat` is getting triggered on page load, which autoFocuse's `ComposerWithSuggestions` component.

### What changes do you think we should make in order to solve the problem?
- We can first get the transaction for current reportID in `ComposerWithSuggestions` component.
```js
transaction: {
    key: ({reportID}) => {
        const report = ReportUtils.getReport(reportID);
        const parentReportAction = ReportActionsUtils.getParentReportAction(report);
        const transactionID = lodashGet(parentReportAction, ['originalMessage', 'IOUTransactionID'], 0);
        return `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`;
    },
},
```

- Then we can check if the transaction is distance request or not.
```js
const isDistanceRequest = TransactionUtils.isDistanceRequest(transaction);
```

- Then we update the `shouldAutoFocus` constant and check if its not a distance request.

https://github.com/Expensify/App/blob/1c8eaa7ca3303c1654b094f6fbaa94bc0c5787fb/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L114

```js
- const shouldAutoFocus = !modal.isVisible && (shouldFocusInputOnScreenFocus || isEmptyChat) && shouldShowComposeInput;
+ const shouldAutoFocus = !isDistanceRequest && !modal.isVisible && (shouldFocusInputOnScreenFocus || isEmptyChat) && shouldShowComposeInput;
```

### What alternative solutions did you explore? (Optional)
- N/A

**Comment by crisps0914** [comment]
Hi
I want to contribute to Expensify.

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't active keyboard automatically for transaction threads

### What is the root cause of that problem?
As we are currently focusing the input whenever the report chat is empty.

### What changes do you think we should make in order to solve the problem?
We can add a condition to prevent focusing the input when the report is a transaction thread even if it is empty. I think the transaction/money request details usually user only checks instead of replying over there so makes sense to avoid focusing the input specific to this scenario.

```
const parentReportAction = ReportActionsUtils.getParentReportAction(report);
```
With `isEmptyChat` add, `(isEmptyChat && !ReportActionsUtils.isTransactionThread(parentReportAction))`
https://github.com/Expensify/App/blob/f77a5a5af54f9c3de6e75baf7046a2831addb15f/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L114

**Comment by JmillsExpensify** [comment]
@Pujan92 we're focusing the input on `chat` reports, which I think makes sense. However, I'm not sure that it makes sense for `expense` report chats and same for the request details view – at least on small screen sizes. @trjExpensify do you agree?

**Comment by trjExpensify** [comment]
I think what @Pujan92 is saying is that any report type that's empty, the composer is focused. Presumably, that's because you're navigating to it for the first time to start a conversation. 

So I agree, for the request transaction thread the composer shouldn't be focused just because there aren't any report comments or system messages in it. You could (and probably more likely) be there to edit a field, just view the request data etc.

**Comment by JmillsExpensify** [comment]
Cool, agreed. @Pujan92 let us know if you have any other questions!

**Comment by sagarguhe** [comment]
Hi, I want to contribute to Expensify

**Comment by sagarguhe** [comment]
Contributor details
Your Expensify account email: sagarguhe@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/sagarguhe

**Comment by trjExpensify** [comment]
>Hi, I want to contribute to Expensify

Hi, @sagarguhe, you can follow the steps in the contributing.md file to join our slack channels to submit bugs and check out the format for proposals.

**Comment by trjExpensify** [comment]
@Santhosh-Sellavel - are you available now to review [this proposal](https://github.com/Expensify/App/issues/26702#issuecomment-1705652067) and progress the PR review?

**Comment by trjExpensify** [comment]
^^ I was just switching out the C+ reviewer. Proposal [here](https://github.com/Expensify/App/issues/26702#issuecomment-1705652067) still needs an internal review.

**Comment by srikarparsi** [comment]
@Pujan92's proposal looks good to me as well, thanks @allroundexperts!

**Comment by Pujan92** [comment]
> @Pujan92 How's the PR going?

@allroundexperts [PR](https://github.com/Expensify/App/pull/26964) is ready for review!

**Comment by JmillsExpensify** [comment]
@allroundexperts can you please fill out the BZ checklist?

**Comment by JmillsExpensify** [comment]
Payment summary:

- Issue reporter: N/A
- Contributor: @Pujan92 $500
- Contributor+: @allroundexperts $500

**Comment by JmillsExpensify** [comment]
Offer sent to @Pujan92. 

$500 payment approved for @allroundexperts based on summary above.

**Comment by JmillsExpensify** [comment]
> Based on my calculations, the pull request did not get merged within 3 working days of assignment. Please, check out my computations here:
> 
> * when @Pujan92 got assigned: 2023-09-06 07:35:45 Z
> * when the PR got merged: 2023-09-11 08:27:26 UTC
> * days elapsed: 3
> 
> On to the next one 🚀

Oh what about this?

**Comment by JmillsExpensify** [comment]
Sorry not trying to be a stickler, but this is over three days. I'm assuming your point is that it's close enough that we should honor it?

**Comment by JmillsExpensify** [comment]
Ok cool. Yes, that's what I was seeing as well. The script is correct, it's just close. @srikarparsi Do you mind confirming that Expensify was responsible for the delay?

**Comment by JmillsExpensify** [comment]
Also while we wait on that confirmation, @allroundexperts do you mind completing the BZ checklist? Thanks!

**Comment by JmillsExpensify** [comment]
Great catch! Ok, I'll remove the hold for the payment for now.

**Comment by Pujan92** [comment]
Ohh, I wasn't aware of the revert.
Raised new PR [here](https://github.com/Expensify/App/pull/29086) @allroundexperts.

**Comment by dylanexpensify** [comment]
@allroundexperts can we get review asap? 🙇‍♂️

**Comment by dylanexpensify** [comment]
bump @allroundexperts

**Comment by JmillsExpensify** [comment]
Looks like this is now with @srikarparsi for review.

**Comment by JmillsExpensify** [comment]
Linked PR is merged, waiting for it to hit production.

**Comment by JmillsExpensify** [comment]
@allroundexperts Do you mind kicking off the BZ checklist and we can get payments issued?

**Comment by JmillsExpensify** [comment]
Payment summary: 

Issue reporter: N/A
Contributor: $500 @Pujan92 
Contributor+: $500 @allroundexperts 

If that looks good to everyone I'll issue payment.

**Comment by JmillsExpensify** [comment]
Ok, thanks! I checked the PR and that makes sense. I've updated the payment summary as a result.

**Comment by JmillsExpensify** [comment]
Alright, paid @Pujan92 the difference, created regression test, and @allroundexperts is paid via NewDot. Closing this issue out.

**Comment by JmillsExpensify** [comment]
$750 payment approved for @allroundexperts based on [this comment](https://github.com/Expensify/App/issues/26702#issuecomment-1823286412).
